pkg/controller/controller_utils.go

```diff
@@ -54,6 +54,13 @@ var (
 	KeyFunc = framework.DeletionHandlingMetaNamespaceKeyFunc
 )
 
+type ResyncPeriodFunc func() time.Duration
```

Comment:
No - it's not for type protection.
The reason I'm doing it is that in some controllers we need more than one resyncPeriod and I want to avoid situation of having many periods to be the same to spread the load over time. That' why I'm injecting the function that is generating random values in production code - the constant values are only in unit tests.